File: android/app/src/main/java/com/sample/locations/LocationData.java
package com.sample.locations;

public class LocationData {
    String timestamp = "0";
    String latitude = "0.0";
    String longitude = "0.0";
    String altitude = "0.0";
    String accuracy = "0.0";
    String speed = "0.0";

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public String getLatitude() {
        return latitude;
    }

    public void setLatitude(String latitude) {
        this.latitude = latitude;
    }

    public String getLongitude() {
        return longitude;
    }

    public void setLongitude(String longitude) {
        this.longitude = longitude;
    }

    public String getAltitude() {
        return altitude;
    }

    public void setAltitude(String altitude) {
        this.altitude = altitude;
    }

    public String getAccuracy() {
        return accuracy;
    }

    public void setAccuracy(String accuracy) {
        this.accuracy = accuracy;
    }

    public String getSpeed() {
        return speed;
    }

    public void setSpeed(String speed) {
        this.speed = speed;
    }
}

File: ios/LocationData.m
//
//  LocationData.m
//  sample
//
//  Created by Saqib on 17/08/2023.
//

#import "LocationData.h"

@implementation LocationData

- (NSDictionary *)toDictionary {
    return @{
        @"timestamp": self.timestamp ?: [NSNull null],
        @"latitude": self.latitude ?: [NSNull null],
        @"longitude": self.longitude ?: [NSNull null],
        @"altitude": self.altitude ?: [NSNull null],
        @"accuracy": self.accuracy ?: [NSNull null],
        @"speed": self.speed ?: [NSNull null]
    };
}

@end

